src/main/java/org/folio/support/ExpirationTool.java

```diff
@@ -103,51 +137,61 @@ private static Request changeRequestStatus(Request request) {
+                                                         String tenant, List<Request> requests) {
 
     Map<String, List<Request>> map = requests.stream()
       .collect(Collectors.groupingBy(Request::getItemId));
 
-    return CompositeFuture.all(map.entrySet()
-      .stream()
-      .map(entry -> updateRequestsPositions(vertx, tenant, entry.getValue()))
-      .collect(Collectors.toList()));
+    Future<Void> future = Future.succeededFuture();
+    for (Map.Entry<String, List<Request>> entry : map.entrySet()) {
+      future = future.compose(v -> updateRequestsPositions(conn, vertx, tenant, entry.getValue()));
+    }
+
+    return future;
   }
 
-  private static Future<CompositeFuture> updateRequestsPositions(Vertx vertx, String tenant, List<Request> requests) {
+  private static Future<Void> updateRequestsPositions(AsyncResult<SQLConnection> conn, Vertx vertx,
+                                                      String tenant, List<Request> requests) {
 
     AtomicInteger pos = new AtomicInteger(1);
+    Future<Void> future = Future.succeededFuture();
 
-    return CompositeFuture.all(requests.stream()
-      .map(request -> updateRequest(vertx, tenant, request.withPosition(pos.getAndIncrement())))
-      .collect(Collectors.toList()));
+    for (Request request : requests) {
+      future = future.compose(v -> updateRequest(conn, vertx, tenant, request.withPosition(pos.getAndIncrement())));
+    }
+
+    return future;
   }
 
 
-  private static Future<CompositeFuture> closeRequests(Vertx vertx, String tenant, List<Request> requests) {
+  private static Future<Void> closeRequests(AsyncResult<SQLConnection> conn, Vertx vertx,
+                                            String tenant, List<Request> requests) {
+
 
-    return CompositeFuture.all(requests.stream()
-      .map(ExpirationTool::changeRequestStatus)
-      .map(ExpirationTool::eraseRequestPosition)
-      .map(request -> updateRequest(vertx, tenant, request))
-      .collect(Collectors.toList()));
+    Future<Void> future = Future.succeededFuture();
+
+    for (Request request : requests) {
+      Request updatedRequest = changeRequestStatus(request).withPosition(null);
+      future = future.compose(v -> updateRequest(conn, vertx, tenant, updatedRequest));
+    }
+
+    return future;
   }
 
-  private static Future<UpdateResult> resetPositionsForOpenRequests(Vertx vertx, String tenant) {
+  private static Future<UpdateResult> resetPositionsForOpenRequests(AsyncResult<SQLConnection> conn, Vertx vertx,
+                                                                    String tenant) {
 
-    String fullTableName = String.format("%s.%s", PostgresClient.convertToPsqlStandard(tenant), REQUEST_TABLE);
+    String fullTableName = format("%s.%s", PostgresClient.convertToPsqlStandard(tenant), REQUEST_TABLE);
 
-    String query = String.format("UPDATE %1$s SET jsonb = jsonb - 'position' WHERE " +
+    String sql = format("UPDATE %1$s SET jsonb = jsonb - 'position' WHERE " +
```

Comment:
This seems to reset the positions for all open requests no matter if there was an expired request for that item.

Am I following this correctly?

Reply:
Yes, you are. Thank you for this point. I have already made corresponding changes.